app/src/main/java/de/westnordost/streetcomplete/osm/cycleway/CyclewayParser.kt

```diff
@@ -94,6 +94,7 @@ private fun parseCyclewayForSide(
     val cycleway = tags[cyclewayKey]
     val cyclewayLane = tags["$cyclewayKey:lane"]
     val isSegregated = tags["$cyclewayKey:segregated"] != "no"
+    val isCyclingOkOnSidewalk = tags["sidewalk$sideVal:bicycle"] == "yes" || tags["sidewalk:both:bicycle"] == "yes"
```

Comment:
`sidewalk:<side>:bicycle=designated` has also some use. But what does it mean?

Probably the same as `cycleway:<side>=track` + `cycleway:<side>:segregated=no`  (<img src="https://upload.wikimedia.org/wikipedia/commons/0/08/Zeichen_240_-_Gemeinsamer_Fu%C3%9F-_und_Radweg%2C_StVO_1992.svg" height="32"/>)?

Or just the same as `sidewalk:<side>:bicycle=yes` (<img src="https://upload.wikimedia.org/wikipedia/commons/0/04/Zusatzzeichen_1022-10_-_Radfahrer_frei%2C_StVO_1992.svg" height="32"/>)? Some mappers consider `designated` to mean "it is signed that way" rather than "it is primarily intended for".

**Four options how to deal with this tagging:**
1.  (**_currently_**:) ignore that value -> interpret it as `NONE`
2. consider this as `SIDEWALK_OK` (<img src="https://upload.wikimedia.org/wikipedia/commons/0/04/Zusatzzeichen_1022-10_-_Radfahrer_frei%2C_StVO_1992.svg" height="32"/>)
3. consider this as `SIDEWALK_EXPLICIT`. (<img src="https://upload.wikimedia.org/wikipedia/commons/0/08/Zeichen_240_-_Gemeinsamer_Fu%C3%9F-_und_Radweg%2C_StVO_1992.svg" height="32"/>)
4. consider this as `INVALID` due to the above ambiguity

I... tend towards considering it invalid, i.e. force users to choose one definite option, due to it being entirely undocumented and ambiguous. I.e. in all places where it is tagged like that, data consumers (routers, editors, visualizers, ...) will have the above options to deal with this, resulting in somewhat inconsistent behavior how this data is treated and this is certainly undesirable.

Tagging @tordans as the main developer of the most detailed cycling map and analysis tool to date (Radverkehrsatlas), @Phyks and @Florimondable as the authors of the Cyclosm cartocss style and @matkoniecz generally for his opinion on tagging.

Reply:
`sidewalk:both:bicycle` ? I would 

I would consider it as  equivalent of `bicycle=yes` on `highway=footway footway=sidewalk` 

So meaning for me would be `SIDEWALK_OK` - but I have no very high confidence it would be shared.

But if that is ambiguous then the same applies to `highway=footway footway=sidewalk bicycle=yes` anyway and to disambig it - one would need to tag specific traffic sign


Note that current use of `sidewalk:both:bicycle` is quite low - see https://overpass-turbo.eu/s/1Laf

Reply:
Read again. This is about `=designated`.

Reply:
From my gut feeling, I would interpret `sidewalk:<side>:bicycle=designated` as someone trying to tag `SIDEWALK_EXPLICIT`, because they seem to try to put the tagging you would give to a standalone path (`highway=path` + `foot=designated` + `bicycle=designated`) to the sidewalk by prefixing those tags with `sidewalk:<side>:`

So my Idea would be: 

- Interpret `sidewalk:<side>:foot=designated` + `sidewalk:<side>:bicycle=designated` as `SIDEWALK_EXPLICIT`
- Interpret only `sidewalk:<side>:bicycle=designated` as `INVALID`, because we can't be 100% sure what is meant

Reply:
> From my gut feeling, I would interpret `sidewalk:<side>:bicycle=designated` as someone trying to tag `SIDEWALK_EXPLICIT`, because they seem to try to put the tagging you would give to a standalone path (`highway=path` + `foot=designated` + `bicycle=designated`) to the sidewalk by prefixing those tags with `sidewalk:<side>:`
> 
> So my Idea would be:
> 
> * Interpret `sidewalk:<side>:foot=designated` + `sidewalk:<side>:bicycle=designated` as `SIDEWALK_EXPLICIT`
> * Interpret only `sidewalk:<side>:bicycle=designated` as `INVALID`, because we can't be 100% sure what is meant

Implemented it like this, for now.

Reply:
> `sidewalk:<side>:foot=designated`

That tag doesn't make sense to me. `sidewalk` already implies it being designated for pedestrians.

Reply:
Hm.... you're right. I guess we should go with `INVALID` then for this case to be safe.

Reply:
I guess "valid" tagging for cycleway should always hinge on `cycleway=` beeing present? So, if no `cycleway=` tag is present at all, but `sidewalk:right:bicycle=yes` & `sidewalk:right:bicycle:signed=yes` is present, should it evaluate to `SIDEWALK_OK` or to `null`? I guess it is `null` in this case, as we want to force the user to make _some_ decision for `cycleway`.

Same with  `sidewalk:<side>:bicycle=designated`. In othwer words, should the check for theese tags be placed as a condition for `"no"` or for `null`?

Reply:
Does your argument not work for just `sidewalk:<side>:bicycle=designated`, without requiring `sidewalk:<side>:foot=designated`? (Looked at taginfo, some sidewalks are indeed tagged like that, but I'd just say that this is superfluous, nothing more?)

Reply:
> Does your argument not work for just `sidewalk:<side>:bicycle=designated`, without requiring `sidewalk:<side>:foot=designated`? (Looked at taginfo, some sidewalks are indeed tagged like that, but I'd just say that this is superfluous, nothing more?)

Yes, after more consideration, I guess it does. I was afraid someone might tag a situation like [this](https://www.google.com/maps/@51.8432515,12.2236736,3a,21.8y,37.79h,91.2t/data=!3m6!1e1!3m4!1s_W3rP2SRvL8O2KTIc6AkvA!2e0!7i16384!8i8192?entry=ttu) where the "sidewalk" is an exclusive cycleway with `sidewalk:<side>:bicycle=designated`, but I guess that would be wrong then. 

If we assume a sidewalk is always `foot=designated`, `sidewalk:<side>:bicycle=designated` would imply `SIDEWALK_EXPLICIT` to me. Still I'm unsure if `cycleway=` needs to be set for this to be valid.

Reply:
> Read again. This is about =designated.

ops

`sidewalk:both:bicycle=designated` would be equivalent of  `bicycle=designated` on `highway=footway footway=sidewalk` 

So meaning for me would be `SIDEWALK_EXPLICIT` - but I have no very high confidence it would be shared.

But if that is ambiguous then the same applies to `highway=footway footway=sidewalk bicycle=designated` anyway and to disambig it - one would need to tag specific traffic sign

Note that current use of `sidewalk:both:bicycle` is quite low - see https://overpass-turbo.eu/s/1Laf

Reply:
Hm, so both of you are of the opinion that `sidewalk:both:bicycle=designated` are equivalent to `cycleway:both=track` + `cycleway:both:segregated=no`.

If this tagging should be supported by StreetComplete, then it would need to be treated in both the parser (i.e. return SIDEWALK_EXPLICIT in that case) and also in the creator (i.e. remove such tagging when another option is selected). + Tests.

Since this was mentioned basically as a side-note of your implementation and has nothing to do directly with it, I leave it up to you if you want to include implementation for that in this PR. It could also be added later.